crates/networking/p2p/rlpx/connection.rs

```diff
@@ -390,6 +460,26 @@ impl<S: AsyncWrite + AsyncRead + std::marker::Unpin> RLPxConnection<S> {
             _ => Err(RLPxError::InvalidState()),
         }
     }
+
+    pub async fn wait_status_msg(&mut self) -> Result<(), RLPxError> {
```

Comment:
I'm not so sure about this.

Disconnect, Ping and Pong should be in the main loop. (and probably Status too). Also if Disconnect is received, it makes no sense to wait for a Status)